### ISSUE TITLE
TMAlarmHandler

### ISSUE THREAD

**Issue by Matteo04052017** [ska-telescope/tmc-prototype#23]
The issue concerns this folder: https://github.com/ska-telescope/tmc-prototype/tree/master/tmcprototype/TMAlarmHandler

Why did you add an executable file into the repository? Where is the code of it?
Why did you create another docker image?
What happen if the device "ska_mid/tm_alarmhandler/tmalarmhandler" is not online when you run the configuration script? 
Why didn't you consider to add a sleep time like in the configuration of the database (https://github.com/ska-telescope/tmc-prototype/blob/master/tmcprototype/addDevicesAndProperties.py)? 

Please let me know if you have any doubts.

**Comment by jyotinpo** [comment]
Hi @Matteo04052017,
My understanding of your questions are inline as comments:

> Why did you add an executable file into the repository? Where is the code of it?

- It was an early stage where we were getting our hands dirty with Dockerfiles. The source file is already available in the [Github repository](https://github.com/ELETTRA-SincrotroneTrieste/alarmhandler) and also at this [link](https://www.tango-controls.org/developers/front-end/#alarm-system). However, we are not using the one that is available in the Github repository. We wanted to avoid issues by mapping to the latest version always. And hence, we compiled the AlarmHandler code in the local system and tested it. Once tested, we wanted to use the same throughout the prototype development. Plus the `tmc-prototype` is in Python while the AlarmHandler code was in C++ which adds complexity to use C++ TANGO docker image. If we use source file, it adds extra compiling steps in the pipeline every time unless we create a docker image which we already did. I believe we can keep it as it is for now. If any of these approaches need improvements, one can take it up in the upcoming PIs if any additional value is generated.

> Why did you create another docker image?

Since AlarmHandler was perceived as a probable solution to implement SKAAlarmHandler, we thought that it will be useful to create a docker image of it so that other Agile teams can also benefit them for their own Alarm handling purpose.

> What happen if the device "ska_mid/tm_alarmhandler/tmalarmhandler" is not online when you run the configuration script?

It was the issue when we were using the custom scripts. Once `tmc-prototype` started using [dsconfig](https://github.com/MaxIV-KitsControls/lib-maxiv-dsconfig), I don't think this is an issue anymore. The configuration can be added into the database even if the `tmalarmhandler` is not running.

> Why didn't you consider to add a sleep time like in the configuration of the database (https://github.com/ska-telescope/tmc-prototype/blob/master/tmcprototype/addDevicesAndProperties.py)?

I don't think this is an issue anymore after using [dsconfig](https://github.com/MaxIV-KitsControls/lib-maxiv-dsconfig) tool.

> Consider also that the alarm configuration script if an alarm already exists.

Since [dsconfig](https://github.com/MaxIV-KitsControls/lib-maxiv-dsconfig) is idempotent, even if the alarm configuration exists already it won't result in any issue or exception as shared in the post.

Let me know if you have any comments/suggestions. We don't have any outstanding items/issues for now related to AlarmHandler, so are we good to close this issue?

**Comment by Matteo04052017** [comment]
What is related to the configuration is solved with the dsconfig.

But you cannot keep a binary file into the repository. Since there is a repository, you could generate the docker image for it and put it into nexus. Once is there, you can just use it. This is what I did for the dsconfig. What do you think?

**Comment by jyotinpo** [comment]
Yeah, I will let @adityadangeska  to take this forward. I think now that we have a TMAlarmHandler image already on Nexus, the binary and other stuffs for TMAlarmHandler from the `tmc-prototype` repository can be removed. One can simply use the image to spawn TMAlarmHandler container. (Unless Nexus repo doesn't crash and all images are lost, then we may have to build and push again.  :rofl: )
